lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3642,9 +3642,34 @@ namespace {
 
       result->setHasUnreferenceableStorage(hasUnreferenceableStorage);
 
-      if (cxxRecordDecl)
+      if (cxxRecordDecl) {
         result->setIsCxxNonTrivial(!cxxRecordDecl->isTriviallyCopyable());
 
+        SmallVector<SubscriptDecl *, 1> subscripts;
+        for (auto &subscriptInfo : Impl.cxxSubscripts) {
+          auto declAndParameterType = subscriptInfo.first;
+          if (declAndParameterType.first != result)
+            continue;
+
+          auto getterAndSetter = subscriptInfo.second;
+          auto subscript = makeSubscript(getterAndSetter.first,
+                                         getterAndSetter.second);
+          subscripts.push_back(subscript);
```

Comment:
I'm not against adding the subscripts in order. But I am curious if there was some reason this needed to be done, or is it just a "cosmetic" thing? 

Additionally, if it would be OK to just match the order in which they are defined in C++, we could maybe make `cxxSubscripts` a `MapVector` and then get rid of the sorting logic. But this is a super small thing and, as I said, I'm not opposed to just using what's here now.

Reply:
Keeping them in order is pretty much only for cosmetics & to make testing of the module interface easier.
I've switched `cxxSubscripts` to `MapVector`, so now we don't need to sort anything here. Thanks for the tip!

Reply:
Awesome, thanks! Still trying to find someone to review the other stuff...